rotkehlchen/csv_exporter.py

```diff
@@ -139,7 +143,14 @@ def reset(self) -> None:
             self.defi_events_csv: List[Dict[str, Any]] = []
             self.ledger_actions_csv: List[Dict[str, Any]] = []
             self.all_events_csv: List[Dict[str, Any]] = []
-            self.all_events = []
+        self.all_events = []
+        self.report_id = 0
```

Comment:
Shouldn't this have a special value? Bellow you call `get_events` but it can happen that no events are stored in the cache right? and then you should query them -> store them -> and finally return them?

Reply:
The way it is currently implemented in this draft, the report queries and computes as before with process_history, and then in the return value, it calls get_events (relying on report_id being initialized in the same call and also events created as they would have normally been added to the csvexporter.all_events List[Dict[str, Any]], but now as database records.

Still remaining are to capture the metadata on the pnl_reports table (the profit_loss totals), then refactor the return value to come from the database instead of memory also with prev next API uris to help the frontend with paging after initially loading 1 page of events.

My vision is that the Generate button will always run processing, but that before clicking generate an index of prior reports will be shown with load actions.

generate will insert and retrieve data from the new database, and load will forgo insertion and only retrieve data from the database.

When viewing a report there will be a back button or [x] to get to the report history/generation screen